homeassistant/components/totalconnect/alarm_control_panel.py

```diff
@@ -58,81 +86,118 @@ def unique_id(self):
+    def state(self):
         """Return the state of the device."""
-        self._client.get_armed_status(self._location_id)
         attr = {
             "location_name": self._name,
             "location_id": self._location_id,
-            "ac_loss": self._client.locations[self._location_id].ac_loss,
-            "low_battery": self._client.locations[self._location_id].low_battery,
-            "cover_tampered": self._client.locations[
-                self._location_id
-            ].is_cover_tampered(),
+            "partition": self._partition_id,
+            "ac_loss": self._location.ac_loss,
+            "low_battery": self._location.low_battery,
+            "cover_tampered": self._location.is_cover_tampered(),
             "triggered_source": None,
             "triggered_zone": None,
         }
 
-        if self._client.locations[self._location_id].is_disarmed():
+        if self._partition.arming_state.is_disarmed():
             state = STATE_ALARM_DISARMED
-        elif self._client.locations[self._location_id].is_armed_night():
+        elif self._partition.arming_state.is_armed_night():
             state = STATE_ALARM_ARMED_NIGHT
-        elif self._client.locations[self._location_id].is_armed_home():
+        elif self._partition.arming_state.is_armed_home():
             state = STATE_ALARM_ARMED_HOME
-        elif self._client.locations[self._location_id].is_armed_away():
+        elif self._partition.arming_state.is_armed_away():
             state = STATE_ALARM_ARMED_AWAY
-        elif self._client.locations[self._location_id].is_armed_custom_bypass():
+        elif self._partition.arming_state.is_armed_custom_bypass():
             state = STATE_ALARM_ARMED_CUSTOM_BYPASS
-        elif self._client.locations[self._location_id].is_arming():
+        elif self._partition.arming_state.is_arming():
             state = STATE_ALARM_ARMING
-        elif self._client.locations[self._location_id].is_disarming():
+        elif self._partition.arming_state.is_disarming():
             state = STATE_ALARM_DISARMING
-        elif self._client.locations[self._location_id].is_triggered_police():
+        elif self._partition.arming_state.is_triggered_police():
             state = STATE_ALARM_TRIGGERED
             attr["triggered_source"] = "Police/Medical"
-        elif self._client.locations[self._location_id].is_triggered_fire():
+        elif self._partition.arming_state.is_triggered_fire():
             state = STATE_ALARM_TRIGGERED
             attr["triggered_source"] = "Fire/Smoke"
-        elif self._client.locations[self._location_id].is_triggered_gas():
+        elif self._partition.arming_state.is_triggered_gas():
             state = STATE_ALARM_TRIGGERED
             attr["triggered_source"] = "Carbon Monoxide"
-        else:
-            logging.info("Total Connect Client returned unknown status")
-            state = None
 
         self._state = state
         self._extra_state_attributes = attr
```

Comment:
It's a bit dangerous to rely on the state property to update the extra attributes. Then we're relying on a detail that is the order and conditions of the property calls in the base entity class during the state update.